### ISSUE TITLE
Can we use horovod to calculate ordinary values?

### ISSUE THREAD

**Issue by tegulia** [uber/horovod#113]
Horovod can easily calculate the gradient of tensorflow. Can we use horovod to calculate ordinary values?For example:
-- | --

    import horovod.tensorflow as hvd
    import numpy as np
    hvd.init()
    hvd_r=int(hvd.rank())
    #each process compute a small part of something and then compute the average etc.
    test_array=np.random.rand(100,100,100)
    #compute a small part
    x=np.mean(test_array[hvd_r*10:(hvd_r+1)*(10),:,:])
    #compute the average for all processes
    y=hvd.sth(x)
    #only one process print out the result
    if(hvd_r==0):
        print("mean of the big array is %f"%y)

**Comment by alsrgv** [comment]
@tegulia, Keras version has functions to do an allreduce on numpy arrays:

```python
import horovod.keras as hvd
import numpy as np
hvd.init()
hvd_r=int(hvd.rank())
#each process compute a small part of something and then compute the average etc.
test_array=np.random.rand(100,100,100)
#compute a small part
x=np.mean(test_array[hvd_r*10:(hvd_r+1)*(10),:,:])
#compute the average for all processes
y=hvd.allreduce(x)
#only one process print out the result
if(hvd_r==0):
    print("mean of the big array is %f"%y)
```

If you want to do general-purpose MPI, you may want to look into [mpi4py](https://mpi4py.readthedocs.io/en/stable/).

**Comment by tegulia** [comment]
that is great! Thanks

**Comment by alsrgv** [comment]
I'll close this issue as it appears to be resolved.
